codenames/guessers/learned_guesser.py

```diff
@@ -0,0 +1,77 @@
+from typing import List
+from overrides import overrides
+
+import torch
+from torch.distributions import Categorical
+
+from codenames.guessers.guesser import Guesser
+from codenames.guessers.policy.guesser_policy import GuesserPolicy
+from codenames.embedding_handler import EmbeddingHandler
+from codenames.utils import game_utils as util
+
+
+class LearnedGuesser(Guesser):
+    def __init__(self,
+                 board: List[str],
+                 embedding_handler: EmbeddingHandler,
+                 policy: GuesserPolicy,
+                 learning_rate: float) -> None:
+        super(LearnedGuesser, self).__init__(board, embedding_handler)
+        self.policy = policy
+        self.guess_history = None
+        self.guess_log_probs = None
+        self.optimizer = torch.optim.Adam(policy.parameters(), lr=learning_rate)
+
+    @overrides
+    def guess(self,
+              clue: str,
+              count: int,
+              game_state: List[int],
+              current_score: int) -> List[str]:
+        # Get vectors for clues and options
+        clue_vector = self.embedding_handler.get_word_vector(clue)
+        if clue_vector is None:
+            return []
```

Comment:
is there a valid situation when this might happen ? If not, lets throw an exception here instead of silently failing ?

Reply:
This ensures the guesser does not guess anything if the clue is OOV. That's the expected behavior.